addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1775,20 +1772,95 @@ options.registry.Carousel = options.Class.extend({
      *
      * @private
      */
-    _addSlide() {
+    async _addSlide() {
+        this.options.wysiwyg.odooEditor.historyPauseSteps();
         const $items = this.$target.find('.carousel-item');
         this.$controls.removeClass('d-none');
         const $active = $items.filter('.active');
         this.$indicators.append($('<li>', {
             'data-target': '#' + this.$target.attr('id'),
-            'data-slide-to': $items.length,
         }));
         this.$indicators.append(' ');
         // Need to remove editor data from the clone so it gets its own.
         $active.clone(false)
             .removeClass('active')
             .insertAfter($active);
-        this.$target.carousel('next');
+        await this._slide("next");
+        this.options.wysiwyg.odooEditor.historyUnpauseSteps();
+    },
+    /**
+     * Slides the carousel in the given direction.
+     *
+     * @private
+     * @param {String|Number} direction the direction in which to slide:
+     *     - "prev": the previous slide;
+     *     - "next": the next slide;
+     *     - number: a slide number.
+     * @returns {Promise}
+     */
+    _slide(direction) {
+        this.trigger_up("disable_loading_effect");
+        let _slideTimestamp;
+        this.$target.one("slide.bs.carousel", () => {
+            _slideTimestamp = window.performance.now();
+            setTimeout(() => this.trigger_up('hide_overlay'));
+        });
+        this.$target.one("slid.bs.carousel", () => {
+            // slid.bs.carousel is most of the time fired too soon by bootstrap
+            // since it emulates the transitionEnd with a setTimeout. We wait
+            // here an extra 20% of the time before retargeting edition, which
+            // should be enough...
+            const _slideDuration = (window.performance.now() - _slideTimestamp);
+            setTimeout(() => {
+                this.trigger_up("activate_snippet", {
+                    $snippet: this.$target.find(".carousel-item.active"),
+                    ifInactiveOptions: true,
+                });
+                this.$target.trigger("active_slide_targeted"); // TODO remove in master: kept for stable.
```

Comment:
There seems to be code using it a few lines below so I don't think this comment is necessary/right

Reply:
I was planning to remove the code below in master too, by making the `slid` listener resolve a promise to make the code simpler/shorter :slightly_smiling_face: 
I thought just saying it on this line would make it clear that it would be removed too, since it is right below 
I will add a comment above it too then :+1:

Reply:
I mean there are two possibilities IMO:

1. You think this code "active_slide_targeted" can be improved by something else: you add a comment saying something like "TODO in master this could be replaced by ...".

2. You think this code "active_slide_targeted" can be improved by something else: you do that something else here directly (provided it is stable enough), but you keep the event trigger for compatibility -> and then you add a comment like you did

So either you comment a line that is now useless the way you did... or you comment why it can be removed in master because _currently_, it's not "kept for compatibility" :wink: 

I know you were about to remove the related code in master too but if I come back later in 15.0 (or if someone from the community does), I should not have to go to master to understand :wink:

Reply:
I will do the second option, as this event was clearly used only for the rest of `remove_slide` to execute, but since the sliding is now awaited, it is now useless
I will keep the trigger in case a custo was using it :slightly_smiling_face: